Sources/OpenAPIKit30/Schema Object/JSONSchemaContext.swift

```diff
@@ -582,9 +582,10 @@ extension JSONSchema {
         ///     is determined by looking at its properties'
         ///     required Bool.
         public var requiredProperties: [String] {
-            return Array(properties.filter { (_, schemaObject) in
-                schemaObject.required
-            }.keys).sorted()
+            properties
+                .filter { _, schema in schema.required }
+                .map { $0.key }
+                .sorted()
```

Comment:
Updated. Yeah, I forgot there is also `Dereferenced `. If its sole purpose is to model "can't be a reference" scenario, I'd personally consider removing it. As a user, it was more of an annoyance to deal with because I had to convert it back to `JSONSchema` everywhere, which is admittedly very easy:

```swift
et schema: JSONSchema
switch value.schema {
case .a(let reference):
    schema = try reference.dereferenced(in: spec.components).jsonSchema
case .b(let value):
    schema = value
}
```

Duplicates don't bother me too much. I'm not planning to support 2.0. But now I needed if I needed to support both 2.0 and 3.0 in my generator, how would that work? I'm assuming OpenAPIKit30 only works with 3.0?

Reply:
I never added support for Swagger/OpenAPI 2.0 to OpenAPIKit, so any support for 2.0 unfortunately would need to be parsed by you and turned into types from OpenAPIKit. There's nothing you can represent in 2.0 that cannot be represented in 3.0, but there are definitely notable differences.

I at least plan to make it much easier to support both 3.0 and 3.1 by creating a bridge module so that either can be parsed and then turned into the other.